### ISSUE TITLE
Telemetry API: deprecate logging configs in MeshConfig

### ISSUE THREAD

**Issue by zirain** [istio/istio#37751]
right now we have two way to set logging:
- `accessLogFile` `accessLogFormat` `accessLogEncoding` in `MeshConfig`
- [Telemetry API](https://istio.io/latest/docs/reference/config/telemetry/)

IMHO, Telemetry API is the first class way to this, let's make a plan to deprecate the legacy way.

**Describe alternatives you've considered**

- mark those configs as deprecated
- add warn in `istioctl analyze`
- verify all doc on `https://istio.io`

@howardjohn @douglas-reid WDYT?

**Comment by howardjohn** [comment]
I would recommend we move everything over (docs, examples, tests, etc) to recommending it and then after a while mark it as deprecated. This is probably the most configured option in Istio so would be good to take our time here.

Do we plan to promote Telemetry to beta first?

**Comment by douglas-reid** [comment]
> Do we plan to promote Telemetry to beta first?

I think we need to approach Beta carefully. I'm worried we don't have enough user feedback yet. Moving everything over would be the next step to getting that feedback. Hopefully, via IstioCon and other venues, we can get broader feedback on the API, as well.

**Comment by zirain** [comment]
>  I would recommend we move everything over (docs, examples, tests, etc) to recommending it and then after a while mark it as deprecated. 

I think it will take 1~2 year(maybe more) to deprecate the `accessLogFile`.

maybe we can start from an announcement like [this](https://istio.io/latest/blog/2021/upcoming-networking-changes/) do.

**Comment by emalihin** [comment]
If I am not mistaken, [Telemetry API for Access Logging](https://istio.io/latest/docs/reference/config/telemetry/#AccessLogging) doesn't expose any of the mentioned configuration options (file, format, encoding) yet?

**Comment by zirain** [comment]
> 

you'd better config them on `ExtensionProvider`, there's the current implement.

**Comment by howardjohn** [comment]
It is configured here:
https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-EnvoyFileAccessLogProvider

**Comment by zirain** [comment]
this's no need after https://github.com/istio/istio/pull/39606.

**Comment by howardjohn** [comment]
IMO this is still desired; we want 1 way to configure things long term

**Comment by zirain** [comment]
keep this for built-in `envoy` provider is easier than deprecated
there's too much field marked as deprecated but not removed in remove
maybe one day this topic will bring up again.

**Comment by howardjohn** [comment]
Deprecating but not removing is a good thing IMO. We can't remove
things;that breaks compatibility. But deprecating them leads users to the 1
right way to do things.

**Comment by zirain** [comment]
> We can't remove things that breaks compatibility. 

Generally, I agree with it. but still like have something like [deprecating-a-feature-or-behavior](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior)

IMO, if you're not plan to remove, mark as deprecating seems meaningless.

For people who really interesting in this, feel free to reopen and take over.

**Comment by howardjohn** [comment]
Its not meaningless. When we have 2 ways (or more) to do one thing, users get confused. By deprecating all but 1 we provide a clear path forward for users so they don't need to decide between two ways to do things.

**Comment by zirain** [comment]
let's add some warning message after Telemetry API promote to beta?

**Comment by jbilliau-rcd** [comment]
Hey guys, came across this issue when trying to figure out if it's possible to add a custom header created by `RequestAuthentication` outputting a claim and write it the envoy access log, but JUST for one namespace via the `Telemetry` object? We have been using it successfully since Istio 1.15 and now have this request from a user, but I don't need or want to do it mesh-wide.

Any idea if that's possible?

**Comment by howardjohn** [comment]
@jbilliau-rcd you would put the two formats in mesh config, then have each namespace reference the one they want

**Comment by jbilliau-rcd** [comment]
@howardjohn yep, just figured that out 5 min ago, thanks for the response. Would be nice (if possible) to adjust the envoy access log format in the `Telemetry` object, so that we don't have to configure the entire mesh config so a customization one dev wants for their one app. Just a user suggestion for the future!

**Comment by zirain** [comment]
> @howardjohn yep, just figured that out 5 min ago, thanks for the response. Would be nice (if possible) to adjust the envoy access log format in the `Telemetry` object, so that we don't have to configure the entire mesh config so a customization one dev wants for their one app. Just a user suggestion for the future!

there's a proposal, but didn't make an agreement.
